duckduckgo/zeroclickinfo-goodies
# Multimonth calendar view goodie

cc @russellholt 
What does your Instant Answer do?

Show a year view of calendars

What problem does your Instant Answer solve (Why is it better than organic links)?

Full calendar views for a year are hard to find

What is the data source for your Instant Answer? (Provide a link if possible)

Perl Datetime and Javascript moment.js

Are there any other alternative (better) data sources?

It would be cool to add holidays at some point. This data is pretty standard.

What are some example queries that trigger this Instant Answer?
[cal | calendar] month year
[cal | calendar]
[cal | calendar] month
[cal | calendar] year

Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)

Everyone who wants to keep track of time.

Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?

Yes it redoes the current calendar. (Speaking of which we may want to rename this to Calendar)

Which existing Instant Answers will this one supercede/overlap with?

CalendarToday

Are you having any problems? Do you need our help with anything?
-How do I best test a primarily javascript based goodie?
-I need help testing the async moment injection since it's not available externally yet
- Should I still be generating plain text?
- Is the selected day working? I can't see on my end.

What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)
![screenshot 2015-08-19 13 43 37](https://cloud.githubusercontent.com/assets/1807895/9364378/076a4d4e-467a-11e5-8f8c-0f68159d1141.png)
![screenshot 2015-08-19 13 43 24](https://cloud.githubusercontent.com/assets/1807895/9364379/076bdc4a-467a-11e5-88f6-0124518aadf0.png)
![screenshot 2015-08-19 13 42 55](https://cloud.githubusercontent.com/assets/1807895/9364380/076d2f96-467a-11e5-857f-117f6cde38c5.png)

Checklist

Please place an 'X' where appropriate.

- [X] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
Unsure what to test here
- [X] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [X] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [X] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 10

    - Windows 7
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 8
        - [] IE 9
        - [] IE 10

    - Windows XP
        - [] IE 7
        - [] IE 8

    - Mac OSX
        - [X] Google Chrome
        - [] Opera

    - iOS (iPhone)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - iOS (iPad)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - Android
        - [] Firefox
        - [] Native Browser
        - [] Google Chrome
        - [] Opera

https://duck.co/ia/view/calendar_today
